Support DIMMER and RGBLIGHTING Supla channels

alh84001 opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This adds support for `DIMMER` and `RGBLIGHTING` Supla channels. Tested with Zamel SLW-01 (`RGBLIGHTING`) and SLW-02 (`DIMMER`) devices.

`RGBLIGHTING` is exposed to HA as light with HS color mode, as it's more practical with existing utils to support scaling HS values with brightness, than it is to do the same with RGB.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/26850

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

alh84001 committed:
Support DIMMER and RGBLIGTING Supla channels

alh84001 committed:
Some clarifying comments.

alh84001 committed:
Add missing method/property docstrings

alh84001 committed:
Fixing ligth.py main docstring.

alh84001 committed:
Naming updates

alh84001 committed:
Properly order imports.

alh84001 commented:
@epenet this is the lighting part of the initial PR, and its ready for review, so take a look when you get a chance. Thanks!

mib1185 committed:
Add entity name translations to NUT (#90709)

epenet committed:
Use entity name translations in Renault (#90697)

emontnemery committed:
Raise repair issue if OTBR and ZHA are on different channels (#90494)

* Raise repair issue if OTBR and ZHA are on different channels

* Update issues after creating or setting dataset

* Explain impact

* Add link to documentation, adjust language

* Update homeassistant/components/otbr/strings.json

---------

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

emontnemery committed:
Bump pychromecast to 13.0.7 (#90724)

mib1185 committed:
Add entity name translations to AVM Fritz!SmartHome (#90707)

* add entity name translation

* sort and capitalize

* adjust tests

* sort entities

epenet committed:
Use entity name translations in 1-wire (#90696)

* Use entity name translations in onewire

* Adjust binary sensors

* Adjust switches

* Cleanup

mib1185 committed:
Add entity name translations to Synology DSM (#90706)

* add entity name translation

* sort strings

* sort and capitalize strings

mib1185 committed:
Add entity name translations to Luftdaten (#90725)

farmio committed:
Use entity name translations in Fronius (#90463)

mib1185 committed:
Add entity name translations to sun (#90732)

ejpenney committed:
Add DHCP discovery to Obihai (#88984)

* Add DHCP discovery to Obihai

* Unique ID is MAC

* Move try blocks, cleanup

* Migrate existing unique_ids

* Use PyObihai to update Unique ID

* Auth then use get_device_mac

* Config flow changes

* Reworking flow according to feedback

* Cleanup

bramkragten committed:
Update frontend to 20230403.0 (#90735)

mib1185 committed:
Add sort list service to Shopping List (#90671)

emontnemery committed:
Make service field filter parameters exclusive (#90728)

TheJulianJES committed:
Restore state for ZHA OnOff binary sensors (#90749)

* Restore state for ZHA OnOff binary sensors

* Let `Motion` extend `Opening`

`Motion` is just a specified version of `Opening` that only changes the device class for some motion sensors.
Since we have more "special code" in the OnOff/Opening sensor now, we also want to make sure that gets applied to `Motion` binary sensors.

* Improve comment and type

* Add test to verify that binary sensors restore last HA state

Djelibeybi committed:
Remove the LIFX sensor update coordinator (#90740)

bieniu committed:
Add entity name translations to NextDNS (#90743)

Add entity name translations

bieniu committed:
Add entity name translations to Tractive (#90738)

Add entity name translations

bdraco committed:
Bump zeroconf to 0.54.0 (#90744)

* Bump zeroconf to 0.54.0

fixes incorrect addresses when the server name changes

changelog: https://github.com/python-zeroconf/python-zeroconf/compare/0.53.0...0.54.0

* fix

bdraco committed:
Prevent overly large event data from being stored in the database (#90747)

This is the same change as #87105 for events

balloob committed:
Run pipeline from audio stream function (#90748)

* Run pipeline from audio stream function

* Fix tests

---------

Co-authored-by: Michael Hansen <mike@rhasspy.org>

balloob committed:
Fix frontend test again (#90754)

bdraco committed:
Bump aiohomekit to 2.6.3 (#90752)

bdraco committed:
Add render count to templates repr (#90753)

epenet commented on homeassistant/components/supla/__init__.py:
```diff
@@ -32,6 +32,8 @@
     "CONTROLLINGTHEGATE": Platform.COVER,
     "CONTROLLINGTHEGARAGEDOOR": Platform.COVER,
     "LIGHTSWITCH": Platform.SWITCH,
+    "DIMMER": Platform.LIGHT,
+    "RGBLIGHTING": Platform.LIGHT,
```
I suggest using alphabetical order here.

epenet commented on homeassistant/components/supla/light.py:
```diff
@@ -0,0 +1,166 @@
+        rgb_color = color_util.rgb_hex_to_rgb_list(hex_color)
+        return color_util.color_RGB_to_hs(rgb_color[0], rgb_color[1], rgb_color[2])
+
+    @property
+    def color_mode(self) -> str | None:
+        """Return the color mode of the light."""
+        return ColorMode.HS
```
Please use shorthand attributes (_attr_color_mode)

epenet commented on homeassistant/components/supla/light.py:
```diff
@@ -0,0 +1,166 @@
+        """Return the color mode of the light."""
+        return ColorMode.HS
+
+    @property
+    def supported_color_modes(self) -> set[str] | None:
+        """Flag supported color modes."""
+        return {ColorMode.HS}
```
Please use shorthand attributes (_attr_supported_color_modes)

epenet commented on homeassistant/components/supla/light.py:
```diff
@@ -0,0 +1,166 @@
+        return ColorMode.BRIGHTNESS
+
+    @property
+    def supported_color_modes(self) -> set[str] | None:
```
Please use shorthand attributes (_attr_supported_color_modes)

epenet commented on homeassistant/components/supla/light.py:
```diff
@@ -0,0 +1,166 @@
+        return min(255, ceil(brightness * 255 / 100.0))
+
+    @property
+    def color_mode(self) -> str | None:
```
Please use shorthand attributes (_attr_color_mode)

epenet commented on homeassistant/components/supla/light.py:
```diff
@@ -0,0 +1,166 @@
+    async_add_entities(entities)
+
+
+class SuplaDimmer(SuplaChannel, LightEntity):
```
```suggestion
class SuplaDimmerEntity(SuplaChannel, LightEntity):
```

epenet commented on homeassistant/components/supla/light.py:
```diff
@@ -0,0 +1,166 @@
+        return {ColorMode.BRIGHTNESS}
+
+
+class SuplaRGBLighting(SuplaChannel, LightEntity):
```
```suggestion
class SuplaRGBLightingEntity(SuplaChannel, LightEntity):
```

epenet reviewed (changes requested):
As per https://github.com/home-assistant/core/pull/90593#issuecomment-1494188592, please create a preliminary PR to rename SuplaChannel to SuplaEntity and move it to a separate entity.py module

emontnemery committed:
Update template environment from the event loop (#90758)

epenet committed:
Add myself to rest code owners (#90770)

starkillerOG committed:
Add async_write_ha_state to Reolink select (#90764)

Add async_write_ha_state to select

raman325 committed:
Add `ms` to units comment for duration sensor device class (#90757)

bdraco committed:
Use slots for discovery dataclasses (#90751)

bdraco committed:
Bump sqlalchemy to 2.0.8 (#90756)

bieniu committed:
Address late review for NextDNS entity name translations (#90771)

mib1185 committed:
Add entity name translations to SMS (#90727)

jbouwh committed:
Fix recovering imap connection triggers re-auth (#90762)

Nardol committed:
Add translations for Nuki entity name and battery critical state attribute (#90772)

* Add translations for Nuki entity name and battery critical state attribute

Signed-off-by: Patrick ZAJDA <patrick@zajda.fr>

* Remove door sensor name

Signed-off-by: Patrick ZAJDA <patrick@zajda.fr>

---------

Signed-off-by: Patrick ZAJDA <patrick@zajda.fr>

hidaris committed:
Preliminary support for Matter cover (#90262)

Preliminary support for Matter cover, curtain tilt support has not been added yet.

MartinHjelmare committed:
Add core as codeowner for stt integration (#90777)

epenet committed:
Use domain constants in rest tests (#90765)

epenet committed:
Rename renault coordinator and base entity (#90760)

Refactor renault coordinator and base entity

MartinHjelmare committed:
Move legacy stt (#90776)

* Move legacy stt to separate module

* Remove case for None as provider

* Add error log for unknown platform

* Add some tests

bdraco committed:
Prevent legacy device tracker from creating hundreds of executor jobs (#90690)

* Prevent legacy device tracker from creating hundreds of executor jobs

The legacy device tracker would create an executor job for
each set of extra state attributes and device name lookup.

For routers this meant hundreds of jobs

* Prevent legacy device tracker from creating hundreds of executor jobs

The legacy device tracker would create an executor job for
each set of extra state attributes and device name lookup.

For routers this meant hundreds of jobs

* tweak

* simplify

alh84001 committed:
Extract Supla base entity into its own file (#90781)

* Extracting Supla base entity

* Fix improper import

* Making Black happy.

* Use set for membership check

* Making ruff happy.

emontnemery committed:
Run prettier on matter fixture (#90784)
